src/components/TodosContainer.jsx

```diff
@@ -0,0 +1,20 @@
+import React from "react";
+import { useSelector } from "react-redux";
+import storage from "../utils/storage";
+import TodoCard from "./TodoCard";
+
+function TodosContainer() {
+  const todos = useSelector(
+    (state) => storage.getTodos("_todos") || state.todo.value
+  );
+
```

Comment:
Basically, `useSelector` is used to get the data from the application store. So instead of getting the todos from the localStorage using the `useSelector`, I am suggesting doing this in the  `initialState` of your `createSlice` function,. By this I mean, you can update your `initialState` to get todos in the `localStorage`, and then refactor your `useSelector` to target the state value like this: 

```suggestion
  const todos = useSelector((state) =>  state.todo.value);
```